app/controllers/reservations_controller.rb

```diff
@@ -304,11 +304,19 @@ def ability_resource
 
   # TODO: you shouldn't be able to edit reservations that have passed or are outside of the cancellation period (check to make sure order has been placed)
   def invalid_for_update?
-    can_edit = @reservation.admin_editable?
-    can_edit &&= @reservation.can_customer_edit? unless current_user.administrator?
     params[:id].to_i != @reservation.id ||
       @reservation.actual_end_at ||
-      !can_edit
+      !editable_by_current_user?
+  end
+
+  def editable_by_current_user?
+    if current_user.administrator? && @reservation.admin_editable?
```

Comment:
That's a good point: I could see this method being replaced with something like `current_ability.can?(:edit, @reservation)`, if the `Ability` file is set up right. I haven't checked. But I'm good with this version.